packages/jellyfish-transaction-builder/__tests__/txn/txn_builder_loan_set_loan_token.test.ts

```diff
@@ -0,0 +1,365 @@
+    }])
+    await testing.generate(1)
+
+    await fundEllipticPair(testing.container, providers.ellipticPair, 10)
+
+    const script = await providers.elliptic.script()
+    const txn = await builder.loans.setLoanToken({
+      symbol: 'Token4',
+      name: 'USD',
+      priceFeedId,
+      mintable: true,
+      interest: new BigNumber(0)
+    }, script)
+    const promise = sendTransaction(testing.container, txn)
+    await expect(promise).rejects.toThrow('DeFiDRpcError: \'LoanSetLoanTokenTx: token \'Token4\' already exists! (code 16)\', code: -26')
+  })
+
+  it('should setLoanToken if name is more than 128 letters', async () => {
+    const priceFeedId = await testing.container.call('appointoracle', [await testing.generateAddress(), [{
+      token: 'Token5',
+      currency: 'USD'
+    }], 1])
+    await testing.generate(1)
+
+    const script = await providers.elliptic.script()
+    const txn = await builder.loans.setLoanToken({
+      symbol: 'Token5',
+      name: 'x'.repeat(129),
+      priceFeedId,
+      mintable: true,
+      interest: new BigNumber(3.5)
+    }, script)
+    await sendTransaction(testing.container, txn)
+    const loanTokenId = calculateTxid(txn)
+    const data = await testing.container.call('listloantokens', [])
+    const index = Object.keys(data).indexOf(loanTokenId) + 1
+    expect(data[loanTokenId].token[index].name).toStrictEqual('x'.repeat(128)) // Only remain first 128 letters.
+  })
+
+  it('should not setLoanToken if priceFeedId does not contain USD price', async () => {
+    const priceFeedId: string = await testing.container.call('appointoracle', [await testing.generateAddress(), [{
+      token: 'Token6',
+      currency: 'SGD'
+    }], 1])
+    await testing.generate(1)
+
+    const script = await providers.elliptic.script()
+    const txn = await builder.loans.setLoanToken({
+      symbol: 'Token6',
+      name: 'Token6',
+      priceFeedId,
+      mintable: true,
+      interest: new BigNumber(4.5)
+    }, script)
+    const promise = sendTransaction(testing.container, txn)
+    await expect(promise).rejects.toThrow(`DeFiDRpcError: 'LoanSetLoanTokenTx: oracle (${priceFeedId}) does not contain USD price for this token! (code 16)', code: -26`)
+  })
+
+  it('should not setLoanToken if priceFeedId is invalid', async () => {
+    const script = await providers.elliptic.script()
+    const txn = await builder.loans.setLoanToken({
+      symbol: 'Token7',
+      name: 'USD',
+      priceFeedId: 'e40775f8bb396cd3d94429843453e66e68b1c7625d99b0b4c505ab004506697b',
+      mintable: true,
+      interest: new BigNumber(5.5)
+    }, script)
+    const promise = sendTransaction(testing.container, txn)
+    await expect(promise).rejects.toThrow('DeFiDRpcError: \'LoanSetLoanTokenTx: oracle (e40775f8bb396cd3d94429843453e66e68b1c7625d99b0b4c505ab004506697b) does not exist or not valid oracle! (code 16)\', code: -26')
+  })
+})
+
+describe('loan.setLoanToken() if mintable is false', () => {
```

Comment:
Create another describe here is not overkill but to make the tests independent from each other.
There is no deleteCollateralToken and getCollateralToken RPCs and

Only like this, the symbol for listloantokens is always be 1
https://github.com/DeFiCh/jellyfish/pull/587/files#diff-ba05900fee1ad772f08d75d4f863d047f043d0ae76dbd8b7bab1a10541862b8eR271

Mintable and interest are optional fields. And this is why I add a separate test for them